### ISSUE TITLE
How gtk.zig was created?

### ISSUE THREAD

**Issue by mikolaszko** [capy-ui/capy#103]
Hi! 
I know that it's not really an issue but there's no discussion board and it's very hard to get a hold of any maintainers' emails or social-media and I really wanted to know whether the gtk.zig file has been coded out from scratch? The file is just massive and I had some gtk experience and I just can't imagine writing eveything there line by line :sweat_smile: 
I am looking into GUI libraries and how they are being built and I'd really appreciate an answer :> 

Btw awesome work and I'll probably build something using capy

**Comment by zenith391** [comment]
Hi,

`gtk.zig` is simply built using Zig's `translate-c` feature. I think I did something along the lines of `zig translate-c -I/usr/include/gtk-4.0/gtk.c > gtk.zig`. Zig can also do it automatically in the source code with
```zig
const c = @cImport({
  @cInclude("gtk-4.0/gtk.h");
});
```
But I chose to generate the `gtk.zig` file anyway because I didn't want to depend on the `gtk.h` header being present on the developer's system, and because currently ZLS generates auto-completions only for `.zig` files, not for `@cImport`ed files.

Hope that answers your question!
